cmd/clusterctl/hack/local-overrides.py

```diff
@@ -0,0 +1,184 @@
+# local repositories (the GitHub repositories clone), and finally stores it in the clusterctl local override folder
+
+# prerequisites:
+
+# - the script should be executed from sigs.k8s.io/cluster-api/ by calling cmd/clusterctl/hack/local-overrides.py
+# - there should be a sigs.k8s.io/cluster-api/clusterctl-settings.json file with the list of provider for which
+#   the local overrides should be generated and the list of provider repositories to be included (on top of cluster-api).
+# {
+#    "providers": [ "cluster-api", "kubeadm", "aws"],
+#    "provider_repos": ["../cluster-api-provider-aws"]
+# }
+# - for each additional provider repository there should be a sigs.k8s.io/<provider_repo>/clusterctl-settings.json file e.g.
+# {
+#   "name": "aws",
+#   "config": {
+#     "componentsFile": "infrastructure-components.yaml",
+#     "nextVersion": "v0.5.0",
+#     "type": "InfrastructureProvider"
+# }
+
+###################
+
+import json
+import subprocess
+import os
+import errno
+
+settings = {}
+
+providers = {
+      'cluster-api': {
+              'componentsFile': 'core-components.yaml',
+              'nextVersion': 'v0.3.0',
+              'type': 'CoreProvider',
+      },
+      'kubeadm': {
+            'componentsFile': 'bootstrap-components.yaml',
+            'nextVersion': 'v0.3.0',
+            'type': 'BootstrapProvider',
+            'configFolder': '/bootstrap/kubeadm/config/default',
+      },
+}
+
+def load_settings():
+    global settings
+    try:
+        settings = json.load(open('clusterctl-settings.json'))
+    except  Exception as e:
+        raise Exception('failed to load clusterctl-settings.json: {}'.format(e))
+
+def load_provider():
+    provider_repos = settings.get('provider_repos', [])
+    for repo in provider_repos:
+        file = repo + '/clusterctl-settings.json'
+        try:
+            provider_details = json.load(open(file))
+            provider_name = provider_details['name']
+            provider_config = provider_details['config']
+            provider_config['repo'] = repo
+            providers[provider_name] = provider_config
+        except  Exception as e:
+            raise Exception('failed to load clusterctl-settings.json from repo {}: {}'.format(repo, e))
+
+def kustomize(args):
+    try:
+        out = subprocess.Popen(args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT)
+
+        stdout, stderr = out.communicate()
+        if stderr is not None:
+            raise Exception('stderr contains: \n{}'.format(stderr))
+
+        return stdout
+    except  Exception as e:
+        raise Exception('failed to run {}: {}'.format(args, e))
+
+def get_home():
+    return os.path.expanduser('~')
+
+def write_local_override(provider, version, components_file, components_yaml):
+    try:
+        home = get_home()
+        overrides_folder = os.path.join(home, 'cluster-api', 'overrides')
```

Comment:
I think we'd want this in a dotfolder, like `.cluster-api`, right?

Reply:
I the CAEP the consensus was for `$HOME/cluster-api/.clusterctl.yaml`
If you are ok I will be happy to change it into `$HOME/.cluster-api/clusterctl.yaml` in a follow-up PR

Reply:
`$HOME/cluster-api` feels very wrong to me. Follow-up is fine, but let's remember to do it 😄

Reply:
Huge +1, the folder should be hidden